src/utp_stream.cpp

```diff
@@ -341,6 +488,9 @@ struct utp_socket_impl
 
 	void set_state(int s);
 
+	packet *acquire_packet(std::uint16_t allocate) { return m_packet_pool->acquire(allocate); }
```

Comment:
in the future, in a different patch, it would be nice to wrap this pointer in a holder with a private constructor that only the packet_pool can invoke, and make it move-only. That way the only way to make it compile is to return the packet to the pool.